contracts/token_wrapper_handler/lib.rs

```diff
@@ -388,7 +388,7 @@ mod token_wrapper_handler {
             resource_id: [u8; 32],
             function_signature: [u8; 4],
             nonce: [u8; 8],
```

Comment:
Noticing that this is wrong, you have it as 8-bytes when it should be 4-bytes. Can you fix @dharjeezy? Can we also simply use `u32` type instead of bytes and when we need bytes encoding to use big-endian.